### DISCUSSION TITLE
multiple groups allocating IPs out of same prefix

### DISCUSSION THREAD

**Discussion by kollross** [netbox-community/netbox#14292] [Help Wanted!]
I have two groups of people (at the moment they are tenants) that I would like them both able to allocate IPs out of a common prefix.   

I had previously went with two different tags "groupA and groupB" that I assigned to the prefix.  I also have constraints that limit visibility of this prefix and IPs to only those two tags.   However when allocating the IP's, the tags are not inherited to the new IP address, which caused the user to not be able to see the IP address they just allocated.   The work around was to pre-allocated all the IP's in the prefix, then assign them the correct tags (and disable the user from deleting IP's so they wouldn't delete the tag).   As you can imagine this was less than ideal. 

I read that Tenants are inherited to the IP from the prefix when they are allocated, so I converted the tag mechanism a Tenant setup, only to find that I can only assign one tenant to a prefix, thus not solving the problem.

I'm just trying to figure is there an actual way to model this in Netbox?  

Note: I can't break up the prefix into small prefixes.

**Comment by candlerb** [comment]
Some ideas, untested:
* Write two permissions/constraints, one allowing tenantA and one allowing tenantB
* Put the tenants in a tenant group, and write the permission to use `tenant__group__name` or similar (but note that each tenant can only belong to one tenant group)

This is one of those problems which might be resolved if Netbox ever gets a direct database relationship from IP address to prefix: #7845
